app/models/contract.rb

```diff
@@ -63,9 +63,14 @@ def self.issued_by(issuer, *ids)
   end
 
   scope :permitted_for, ->(user) {
-    next all unless user.forbidden_some_services?
-
-    where(service_id: user.member_permission_service_ids)
+    case user.permitted_services_status
+    when :none
+      next none
```

Comment:
Actually, now with `case` we can remove `next`, but I'd still keep this as `none`, and not as `public_send`, I think it makes it clearer.